osu.Game.Rulesets.Mania/UI/Column.cs

```diff
@@ -44,6 +44,63 @@ public class Column : ScrollingPlayfield, IKeyBindingHandler<ManiaAction>, IHasA
 
         private readonly GameplaySampleTriggerSource sampleTriggerSource;
 
+        public class ColumnTouchInputArea : Container
+        {
+            private Column column => (Column)Parent;
+
+            private Container hintContainer;
+
+            public ColumnTouchInputArea()
+            {
+                RelativeSizeAxes = Axes.X;
+                Anchor = Anchor.BottomCentre;
+                Origin = Anchor.BottomCentre;
+                Height = 100;
+                InternalChild = hintContainer = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    BorderColour = Color4.Red,
+                    BorderThickness = 5,
+                    Masking = true,
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                hintContainer.Delay(1000).FadeOutFromOne(500, Easing.OutSine);
+            }
+
+            private ManiaInputManager.RulesetKeyBindingContainer keyBindingContainer { get; set; }
+
+            private ManiaInputManager.RulesetKeyBindingContainer getKeyBindingContainer()
+            {
+                return keyBindingContainer ??= (ManiaInputManager.RulesetKeyBindingContainer)((ManiaInputManager)GetContainingInputManager()).KeyBindingContainer;
```

Comment:
This should be obtained once, in `LoadComplete` and stored to a private field. Also the case should probably not be direct, as I'm 99.9% sure this is going to break tests.

Reply:
This is still unresolved. Casts are still direct.